tools/clang/test/HLSLFileCheck/shader_targets/library/inout_struct_mismatch-strictudt.hlsl

```diff
@@ -0,0 +1,36 @@
+// RUN: %dxc -T lib_6_x -default-linkage external -strict-udt-casting %s | FileCheck %s
+
+// CHECK: define <4 x float>
+// CHECK-SAME: main
+// CHECK-NOT: bitcast
+// CHECK-NOT: CallStruct
+// CHECK: ParamStruct
+// CHECK: call void @llvm.lifetime.start
+// CHECK-NOT: bitcast
+// CHECK-NOT: CallStruct
+// CHECK-LABEL: ret <4 x float>
+
+struct ParamStruct {
+  int i;
+  float f;
+};
+
+struct CallStruct {
+  int i;
+  float f;
+};
+
+void modify(inout ParamStruct s) {
+  s.f += 1;
+}
+
+void modify_ext(inout ParamStruct s);
+
+CallStruct g_struct;
+
+float4 main() : SV_Target {
+  CallStruct local = g_struct;
+  modify((ParamStruct)local);
+  modify_ext((ParamStruct)local);
+  return local.f;
+}
```

Comment:
I'm curious how these test variants were chosen. Did you try running it with the flag enabled by default and identified the shaders that failed?

Reply:
Yes, I ran it with the flag enabled by default, then added copies of failing ones using the flag with added explicit casts.